client/extensions/hello-dolly/state/reducer.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * Internal dependencies
+ */
+import { HELLO_DOLLY_NEXT_LYRIC } from './action-types';
+import {
+	ROUTE_SET,
+	SECTION_SET,
+	SITE_SETTINGS_SAVE,
+} from 'state/action-types.js';
+
+export default function lyricIndex( state = 0, action ) {
+	switch ( action.type ) {
+		case ROUTE_SET:
+		case SECTION_SET:
+		case SITE_SETTINGS_SAVE:
+		case HELLO_DOLLY_NEXT_LYRIC:
+			return advanceToNextLyric( state );
+		default:
+			return state;
+	}
+}
+
+function advanceToNextLyric( index ) {
+	return index + 1;
```

Comment:
it's not too silly really @gwwar since it's a safe operation to keep on increasing. it's also safe against certain kinds of bugs where the lyrics change

in general I'm hesitant to store lists and indices separate. one of my suggestions was to store the whole list here and rotate it instead of using an index. I like keeping the lyrics length out of this file. it might be that we picked a fairly difficult data type for creating an example